pyam/core.py

```diff
@@ -1770,17 +1768,20 @@ def _check_rows(rows, check, in_range=True, return_test='any'):
     if not set(check.keys()).issubset(valid_checks):
         msg = 'Unknown checking type: {}'
         raise ValueError(msg.format(check.keys() - valid_checks))
-
-    if 'year' not in rows:
-        rows = rows.copy()
-        rows['year'] = rows['time'].apply(lambda x: x.year)
-
-    where_idx = set(rows.index[rows['year'] == check['year']]) \
+    if 'year' not in rows.index.names:
+        where_idx = set(rows.index[rows.index.get_level_values('time').year
+                                    == check['year']]) \
+        if 'year' in check else set(rows.index)
+        rows = rows.loc[list(where_idx)]
```

Comment:
any particular reason for needing to duplicate this line? (see also proposed change below indenting the same line)